src/main/java/com/suse/salt/netapi/errors/FunctionNotAvailable.java

```diff
@@ -20,7 +20,8 @@ public String getFunctionName() {
     public <T> T fold(Function<FunctionNotAvailable, ? extends T> fnNotAvail,
             Function<ModuleNotSupported, ? extends T> modNotSupported,
             Function<JsonParsingError, ? extends T> jsonError,
-            Function<GenericError, ? extends T> generic
+            Function<GenericError, ? extends T> generic,
+            Function<SaltSSHError, ? extends T> saltSshError
```

Comment:
Very minor comment here: for more consistency as we use `SSH` in the class name (which I like), I would maybe use the same naming for the variable `saltSSHError`, starting with a lower-cased `s`.